frontend/lib/models/filter.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:dartx/dartx.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'filter.freezed.dart';
+
+@freezed
+class Filter<T> with _$Filter<T> {
+  const Filter._();
+
+  const factory Filter({
+    required String name,
+    required String? Function(T) extractOption,
+    required List<({String name, bool value})> options,
+  }) = _Filter<T>;
+
+  factory Filter.fromObjects({
+    required String name,
+    required String? Function(T) extractOption,
+    required List<T> objects,
+  }) {
+    final names = <String>{};
+    for (final artefact in objects) {
```

Comment:
Sorry to be really nitpicking here, but this is a bit confusing as it is not really iterating over artefacts, but any type of object that uses this filter, if you don't mind we can change it to `for (final object in objects)`

Reply:
Ah good catch, I missed this